Buttons für typische Editoraktionen (Fett, Kursiv, Unterstreichen)

NickSchmahl opened the pull request:
In dieser PR habe ich in einer kleinen Toolbar über dem Editor die Knöpfe Paragraph und Delete durch drei Knöpfe (Fett, Kursiv, Unterstreichen) ersetzt. Diese Knöpfe haben auch key bindings bekommen (Strg + B & Strg + I) (welche das final sind, kann man natürlich noch diskutieren). Strg + U für Unterstreichen ist im ACE-Editor leider schon durch "toUpperCase" belegt, mal gucken ob man das noch wechseln kann, ich hab es nicht hinbekommen. 

Ich würde gerne noch schaffen, dass man die Formatierung auch rückgängig machen kann mit dem Knopfdruck wenn der markierte Bereich schon fett usw ist. Dies würde in nächsten Commits folgen.

Mit der Formatierung im Text warte ich noch auf Arbeit von @Nummersechs, weil dieser lokal mit Fettgedrucktem Editor Text und so schon etwas rumgespielt hat und ich das dann übernehmen würde. Wird aber vermutlich nicht mehr Teil dieser PR.

chakuTech committed:
New buttons for bold, italic, underline

chakuTech committed:
New buttons for bold, italic, underline

chakuTech committed:
Making the right selection after surrounding text

chakuTech committed:
Making the right selection after surrounding text

chakuTech committed:
Shrinking the editor size so that no screen is lost

chakuTech committed:
Remove "Delete" and "Paragraph from Actions

chakuTech committed:
Remove "Delete" and "Paragraph from Actions

lassezuengel committed:
Add abstract Pagination component

Allows for different styles and configurability options for pagination.
Easily extendable for future features and stuff.

Can be (re)used for lots of different pages, like the
list of projects on the home page,
the list of users on the admin panel, etc.

lassezuengel committed:
Add abstract Pagination component

Allows for different styles and configurability options for pagination.
Easily extendable for future features and stuff.

Can be (re)used for lots of different pages, like the
list of projects on the home page,
the list of users on the admin panel, etc.

lassezuengel committed:
Improve user list filtering and pagination

Also some minor fixes to the pagination component.

lassezuengel committed:
Improve user list filtering and pagination

Also some minor fixes to the pagination component.

lassezuengel committed:
Add more mock projects and pagination to home page

This commit also adds querying to the pagination component. Namely,
we can now tell the pagination object to change pages. Can be extended
even further, but because the input system already handles automatic
resizing and styling, this should be sufficient.

lassezuengel committed:
Add more docs to Pagination

lassezuengel committed:
Add more docs to Pagination

lassezuengel committed:
Clean up some todos

lassezuengel committed:
Clean up some todos

Nummersechs committed:
Editor highlight and marker (#160)

Added these feauters:

- A custom highlighter #73  to the editor using JS and CSS.
- Updated TOCEntry with marker and annotation.
- Added a button for the user to mark a section as a comment and delete
them.
- Added different gutter annotation (icon on the left side of the
editor) for the annoation type info

---------

Co-authored-by: Nummersechs <stu204034@mail.uni-kiel.de>

Nummersechs committed:
Editor highlight and marker (#160)

Added these feauters:

- A custom highlighter #73  to the editor using JS and CSS.
- Updated TOCEntry with marker and annotation.
- Added a button for the user to mark a section as a comment and delete
them.
- Added different gutter annotation (icon on the left side of the
editor) for the annoation type info

---------

Co-authored-by: Nummersechs <stu204034@mail.uni-kiel.de>

Nummersechs committed:
Delete docs/language.zip (#163)

accidentally added this zip

Nummersechs committed:
Delete docs/language.zip (#163)

accidentally added this zip

respiranto committed:
ltml: Change reference syntax

From `{LABEL}` to `{:LABEL}`.

closes: #132

respiranto committed:
ltml: Change reference syntax

From `{LABEL}` to `{:LABEL}`.

closes: #132

Nummersechs committed:
extracted the toc content in sidebar to its own component

Nummersechs committed:
extracted the toc content in sidebar to its own component

Nummersechs committed:
moved AnnotatedMarker and TOCEntry to Types.purs

Nummersechs committed:
moved AnnotatedMarker and TOCEntry to Types.purs

respiranto committed:
ltml: Add general labeling parser

We removed `:` from the set of acceptable label characters.  This is to
keep parsing simple, and maybe avoid confusing users.

Also,
 - update docs on label and labeling syntax
 - use labeling parser for paragraphs (addresses #135 w.r.t. paragraphs)

respiranto committed:
ltml: Add general labeling parser

We removed `:` from the set of acceptable label characters.  This is to
keep parsing simple, and maybe avoid confusing users.

Also,
 - update docs on label and labeling syntax
 - use labeling parser for paragraphs (addresses #135 w.r.t. paragraphs)

Nummersechs committed:
removed endRow and endCol from AnnotedMarker

Nummersechs committed:
removed endRow and endCol from AnnotedMarker

Nummersechs committed:
applied purs-tidy

Nummersechs committed:
applied purs-tidy

respiranto committed:
backend/ltml: Add sentence start parsing

There is still work to do; notably, we currently accept
* a sentence start token at the end of a paragraph, and
* a paragraph composed of nothing but a sentence start token.

Closes: #103
Closes: #152

Opens: #168
Opens: #169

respiranto committed:
backend/ltml: Add sentence start parsing

There is still work to do; notably, we currently accept
* a sentence start token at the end of a paragraph, and
* a paragraph composed of nothing but a sentence start token.

Closes: #103
Closes: #152

Opens: #168
Opens: #169

respiranto committed:
backend/ltml: Add space token

Previously, space was encoded as `TextLeaf " "`, and we also often
parsed `TextLeaf ""`; instead, we now parse `[Space]` and `[]`,
respectively.

This also better matches Pandoc's AST.

respiranto committed:
backend/ltml: Add space token

Previously, space was encoded as `TextLeaf " "`, and we also often
parsed `TextLeaf ""`; instead, we now parse `[Space]` and `[]`,
respectively.

This also better matches Pandoc's AST.

respiranto committed:
backend/ltml: Simplify word parsing

Previously, the ad-hoc-polymorphic textLeafP function returned a
TextTree and was to always use the TextLeaf constructor.

What we actually would have always wanted was an ad-hoc-polymorphic
function that returns a Text, but that is not possible because the type
class we define it in depends on a type parameter of TextTree.

Instead, we now use Proxy.

Also, we renamed the TextLeaf constructor to Word, which is fitting now
that whitespace is no longer encoded as a TextLeaf.

respiranto committed:
backend/ltml: Simplify word parsing

Previously, the ad-hoc-polymorphic textLeafP function returned a
TextTree and was to always use the TextLeaf constructor.

What we actually would have always wanted was an ad-hoc-polymorphic
function that returns a Text, but that is not possible because the type
class we define it in depends on a type parameter of TextTree.

Instead, we now use Proxy.

Also, we renamed the TextLeaf constructor to Word, which is fitting now
that whitespace is no longer encoded as a TextLeaf.

merliin3007 committed:
Backend: Internal Document API and Commits with Multiple Parents (#175)

This PR adds new functions to the internal Version Control API to work
with documents:
- `getDocument` gets a document by id
- `createDocument` creates a new document for a user group
- `createDocumentCommit` creates a new commit and tries to set it as the
new document head commit. This can fail if the new commit is unrelated
to the currenty set head commit of the document. This might change in
the future, as there are other cases in which it might be sensible to
fail the operation.

Also, commits can now have multiple parent commits. (yeah, this probably
should have been a separate PR)

---------

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Backend: Internal Document API and Commits with Multiple Parents (#175)

This PR adds new functions to the internal Version Control API to work
with documents:
- `getDocument` gets a document by id
- `createDocument` creates a new document for a user group
- `createDocumentCommit` creates a new commit and tries to set it as the
new document head commit. This can fail if the new commit is unrelated
to the currenty set head commit of the document. This might change in
the future, as there are other cases in which it might be sensible to
fail the operation.

Also, commits can now have multiple parent commits. (yeah, this probably
should have been a separate PR)

---------

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Add JSON Instances for Documents (#176)

This PR adds `ToJSON`, `FromJSON` and `ToSchema` instances to `Document`
and `DocumentID`.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Add JSON Instances for Documents (#176)

This PR adds `ToJSON`, `FromJSON` and `ToSchema` instances to `Document`
and `DocumentID`.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Backend VersionControl: Add Missing Doc Comments (#177)

This PR adds missing doc comments to the backends `VersionControl`
module.

---------

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Backend VersionControl: Add Missing Doc Comments (#177)

This PR adds missing doc comments to the backends `VersionControl`
module.

---------

Co-authored-by: Finn Evers <dev@bahn.sh>

chakuTech committed:
Some key presses will be logged in the console

chakuTech committed:
Some key presses will be logged in the console

merliin3007 committed:
Backend VersionControl: Fetch Commit Graph for Document (Internal API) (#178)

This PR adds the function `getCommitGraph` to the internal Version
Control API. This function enables the user to obtain the whole commit
graph for a document from the database. The graph is currently
represented as a vector of commits. The relations between commits are
included in the commits.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Backend VersionControl: Fetch Commit Graph for Document (Internal API) (#178)

This PR adds the function `getCommitGraph` to the internal Version
Control API. This function enables the user to obtain the whole commit
graph for a document from the database. The graph is currently
represented as a vector of commits. The relations between commits are
included in the commits.

Co-authored-by: Finn Evers <dev@bahn.sh>

chakuTech committed:
bold, italic and underscore as keybindings

chakuTech committed:
bold, italic and underscore as keybindings

chakuTech committed:
Keybindings in eigene Datei ausgelagert

chakuTech committed:
Keybindings in eigene Datei ausgelagert

chakuTech committed:
key binding for underscore is bad because its also for uppercase

merliin3007 committed:
Backend VersionControl: Add Web Parameter Instances for DocumentID (#180)

Due to the instances implemented in this PR for `DocumentID`, it is now
possible to use the `DocumentID` e.g., as a query parameter.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Backend VersionControl: Add Web Parameter Instances for DocumentID (#180)

Due to the instances implemented in this PR for `DocumentID`, it is now
possible to use the `DocumentID` e.g., as a query parameter.

Co-authored-by: Finn Evers <dev@bahn.sh>

chakuTech committed:
Use Translator in editor buttons

chakuTech committed:
Use Translator in editor buttons

merliin3007 committed:
Fix Backend Bug (#183)

This PR fixes a bug introduced by me in #175. Just a small typo, forgot
on column in a SQL query. Due to this bug, the server was unable to
start.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Fix Backend Bug (#183)

This PR fixes a bug introduced by me in #175. Just a small typo, forgot
on column in a SQL query. Due to this bug, the server was unable to
start.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Implement ToParamSchema and FromHttpApiData for Version Control IDs (#184)

This PR implementes `ToParamSchema` and `FromHttpApiData` instances for
the id types in the `VersionControl` module (`CommitID`, `NodeID` and
`Hash`). This is required to use these ids in api paths for implementing
a REST wrapper around the internal `VersionControl` api.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Backend VersionControl: Refactor Document API to Behave More Like the Commit API (#185)

This PR aims to maintain a uniform behavior of all internal api
functions of the `VersionControl` module.
All functions of the internal `VersionControl` api behaved the same in
the way that functions creating new database objects always returned the
whole object as a result. Until this PR, the only exception was the
`createDocument` function. In order to avoid confusion, `createDocument`
now behaves like all similar functions of the internal `VersionControl`
api.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Backend VersionControl: Refactor Document API to Behave More Like the Commit API (#185)

This PR aims to maintain a uniform behavior of all internal api
functions of the `VersionControl` module.
All functions of the internal `VersionControl` api behaved the same in
the way that functions creating new database objects always returned the
whole object as a result. Until this PR, the only exception was the
`createDocument` function. In order to avoid confusion, `createDocument`
now behaves like all similar functions of the internal `VersionControl`
api.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Improve Docs of Internal VersionControl API (#186)

This PR improves the documentation of the internal `VersionControl` API.

Co-authored-by: Finn Evers <dev@bahn.sh>

merliin3007 committed:
Improve Docs of Internal VersionControl API (#186)

This PR improves the documentation of the internal `VersionControl` API.

Co-authored-by: Finn Evers <dev@bahn.sh>

respiranto committed:
backend/ltml: Fix sentence start parser

(a) Sentence start tokens must always be succeeded by an element (i.e., must
    not be last element in a given context / parent element).
    See also #168.

(b) No whitespace tokens should be generated directly after a sentence
    start token.  See #174.

This notably extends the MiTree parser combinators, by adding
`MiElementConfig`.

fixes: #168
fixes: #174

respiranto committed:
backend/ltml: Fix sentence start parser

(a) Sentence start tokens must always be succeeded by an element (i.e., must
    not be last element in a given context / parent element).
    See also #168.

(b) No whitespace tokens should be generated directly after a sentence
    start token.  See #174.

This notably extends the MiTree parser combinators, by adding
`MiElementConfig`.

fixes: #168
fixes: #174

respiranto committed:
backend/ltml: Terminate sentence after enum items

See also #147.

respiranto committed:
backend/ltml: Terminate sentence after enum items

See also #147.

Nummersechs commented:
Ich mache momentan nichts mehr am Textformatierung. Das Language Team kann dann später die Regex für das Highlighten noch ergänzen. Ich arbeite aktuell an den Kommentarensektion und dessen Struktur.

MrSubidubi committed:
frontend: Add ACE customizations to Dockerfile (#189)

This PR ensures the ACE customizations are included in the build docker
images.

MrSubidubi committed:
frontend: Add ACE customizations to Dockerfile (#189)

This PR ensures the ACE customizations are included in the build docker
images.

Fab1000 committed:
Render endpoint for HTML and plain text (#179)

This PR adds:
- endpoint `/documents/render/{format}` which takes a plain text and
returns a rendered bytestring
- currently two formats are supported: `html` and plain text (`plain`)
- also slightly refactored the API naming for less redundancy

respiranto committed:
backend/ltml: Revise parser type class hierarchy

respiranto committed:
backend/ltml: Revise parser type class hierarchy

respiranto committed:
backend/ltml: Add sentence continuation

Sentences that would otherwise end can now be continued via `{>}`.  This
was originally only intended for use after enumerations (which otherwise
terminate sentences), but was now implemented more generally, which
happens to be easier, and is deemed to be useful from a user's
perspective.

closes: #147

respiranto committed:
backend/ltml: Add sentence continuation

Sentences that would otherwise end can now be continued via `{>}`.  This
was originally only intended for use after enumerations (which otherwise
terminate sentences), but was now implemented more generally, which
happens to be easier, and is deemed to be useful from a user's
perspective.

closes: #147

NickSchmahl commented:
Alles klar, dann gehe ich da später noch ran :)

chakuTech committed:
Resolve merge conflicts
